moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame.cpp

```diff
@@ -426,6 +427,70 @@ void MotionPlanningFrame::sceneUpdate(planning_scene_monitor::PlanningSceneMonit
     planning_display_->addMainLoopJob(boost::bind(&MotionPlanningFrame::populateCollisionObjectsList, this));
 }
 
+void MotionPlanningFrame::addPrimitiveShape()
+{
+  static const double MIN_VAL = 1e-6;
+  static const std::map<std::string, shapes::ShapeType> SHAPES_MAP = {
+    { "box", shapes::BOX }, { "sphere", shapes::SPHERE }, { "cone", shapes::CONE }, { "cylinder", shapes::CYLINDER }
+  };
+
+  if (!planning_display_->getPlanningSceneMonitor())
+  {
+    return;
+  }
+
+  // get side length
+  double side_length = ui_->shape_size_spin_box->value();
+  if (side_length < MIN_VAL)
+  {
+    QMessageBox::warning(this, QString("Side length value is too small"),
+                         QString("Value needs to be  >= '").append(std::to_string(MIN_VAL).c_str()).append("'"));
+    return;
+  }
+
+  // get selected shape
+  std::string selected_shape = ui_->shapes_combo_box->currentText().toStdString();
+  if (SHAPES_MAP.count(selected_shape) == 0)
```

Comment:
You could also use `auto selected_shape_it =  SHAPES_MAP.find(selected_shape)` and test if it's equal to `SHAPES_MAP.end()` and replace `SHAPES_MAP.at(selected_shape);` with the object it point to